### ISSUE TITLE
.catch({ code: '...'}, ...)

### ISSUE THREAD

**Issue by dantman** [petkaantonov/bluebird#499]
Another idea for catch, a lot of errors node and otherwise have a code attribute which is often very informative.

Being able to write something like:
```js
  .catch({ code: 'ENOENT' }, function(err) {
    consle.log('%s does not exist', err.path);
  })
```

Would make a lot of error catches very specific and very concise.

Using an object notation (like what lodash uses for _.where) would also support things like `{ status: 404 }`.

**Comment by petkaantonov** [comment]
I agree, in the meanwhile you could check out https://github.com/petkaantonov/core-error-predicates

**Comment by benjamingr** [comment]
I personally solve this with a predicate catch and Ramda.

**Comment by kjvalencik** [comment]
I also solve this with predicate catch, but with lodash instead of Ramda. E.g.,

`.catch(_.matches({ code : "ENOENT" }))`

**Comment by petkaantonov** [comment]
Fixed in 3.0 release
